pkg/liqo-controller-manager/external-network/configuration-controller/configuration-controller.go

```diff
@@ -0,0 +1,131 @@
+	if err := r.Get(ctx, req.NamespacedName, configuration); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("There is no configuration %s", req.String())
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf(" %w --> Unable to get the configuration %s", err, req.Name)
+	}
+	events.Event(r.EventsRecorder, configuration, "Processing")
+
+	requeue, err := r.RemapConfiguration(ctx, configuration, r.EventsRecorder)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err = r.UpdateConfigurationStatus(ctx, configuration); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if requeue {
+		events.Event(r.EventsRecorder, configuration, "Waiting for the network to be ready")
+	} else {
+		events.Event(r.EventsRecorder, configuration, "Configuration remapped")
+		err = SetConfigurationConfigured(ctx, r.Client, configuration)
+	}
+	return ctrl.Result{Requeue: requeue}, err
+}
+
+// RemapConfiguration remap the configuration using ipamv1alpha1.Network.
+func (r *ConfigurationReconciler) RemapConfiguration(ctx context.Context, cfg *networkingv1alpha1.Configuration,
+	er record.EventRecorder) (requeue bool, err error) {
+	// Checks if the configuration is already remapped.
+	for _, cidrType := range LabelCIDRTypeValues {
+		network, err := CreateOrGetNetwork(ctx, r.Client, r.Scheme, er, cfg, cidrType)
+		if err != nil {
+			return true, fmt.Errorf(" %w --> Unable to create or get the network %s", err, cfg.Name)
+		}
+		if network.Status.CIDR == "" {
+			return true, nil
+		}
+		ForgeConfigurationStatus(cfg, network, cidrType)
+	}
+	return false, nil
+}
+
+// UpdateConfigurationStatus update the configuration.
+func (r *ConfigurationReconciler) UpdateConfigurationStatus(ctx context.Context, cfg *networkingv1alpha1.Configuration) error {
+	if err := r.Client.Status().Update(ctx, cfg); err != nil {
+		return fmt.Errorf(" %w --> Unable to update the status of the configuration %s", err, cfg.Name)
+	}
+	return nil
+}
+
+// ForgeConfigurationStatus create the status of the configuration.
+func ForgeConfigurationStatus(cfg *networkingv1alpha1.Configuration, net *ipamv1alpha1.Network, cidrType LabelCIDRTypeValue) {
+	if cfg.Status.Remote == nil {
+		cfg.Status.Remote = &networkingv1alpha1.ClusterConfig{}
+	}
+	var cidrNew, cidrOld networkingv1alpha1.CIDR
+	cidrNew = net.Status.CIDR
+	switch cidrType {
+	case LabelCIDRTypePod:
+		cidrOld = cfg.Spec.Remote.CIDR.Pod
+		cfg.Status.Remote.CIDR.Pod = cidrNew
+	case LabelCIDRTypeExternal:
+		cidrOld = cfg.Spec.Remote.CIDR.External
+		cfg.Status.Remote.CIDR.External = cidrNew
+	}
+	klog.Infof("Configuration %s %s CIDR: %s -> %s", client.ObjectKeyFromObject(cfg).String(), cidrType, cidrOld, cidrNew)
+}
+
+// SetupWithManager register the ConfigurationReconciler to the manager.
+func (r *ConfigurationReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).For(&networkingv1alpha1.Configuration{}).Complete(r)
```

Comment:
Can we trigger reconciliations from the onwed networks intead of using the requeue flag?